src/rust/ide/src/double_representation/alias_analysis.rs

```diff
@@ -65,26 +72,257 @@ pub struct IdentifierUsage {
 // === Analysis ===
 // ================
 
+
+// === Helper Datatypes ===
+
+/// Says whether the identifier occurrence introduces it into scope or uses it from scope.
+#[allow(missing_docs)]
+#[derive(Clone,Copy,Debug,Display,PartialEq)]
+pub enum OccurrenceKind { Used, Introduced }
+
+/// If the current context in the AST processor is a pattern context.
```

Comment:
I'd like to have a reference to a document which describes what pattern context is.

Reply:
Will add TODO comment, to be fixed when #349 lands.